packages/client/src/pages/Forum/TopicList.tsx

```diff
@@ -1,113 +1,59 @@
-import { FC, useState } from 'react';
-import { Link, useParams } from 'react-router-dom';
+import { FC, useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+
+import { forumApi } from '@api/forumApi';
 
 import { NewTopic } from './components/NewTopic';
+import { TopicItem } from './components/TopicItem';
 import { ITopicTypes } from './types';
 
 export const TopicList: FC = () => {
-  const testTopicList = [
-    {
-      title: 'Некоректное начисление бонусов за матч',
-      author: 'FlexK1ngN1TTV',
-      answers: 1,
-      lastPublic: {
-        time: '4 часа назад',
-        author: 'Техническая поддержка',
-      },
-      id: '8bxnesczx4',
-    },
-    {
-      title: 'Баг в меню, не работает кнопка играть',
-      author: 'Falepton',
-      answers: 5,
-      lastPublic: {
-        time: '5 часа назад',
-        author: 'Техническая поддержка',
-      },
-      id: 'ie9q2fkvu8',
-    },
-    {
-      title: 'Как пройти финальный уровень?',
-      author: 'Dalsem',
-      answers: 2,
-      lastPublic: {
-        time: '10 часов назад',
-        author: 'Пользователь: Falepton',
-      },
-      id: 'afwnh0lmjv',
-    },
-    {
-      title: 'Персонаж телепортируется во время игры',
-      author: 'Copatich',
-      answers: 0,
-      lastPublic: {
-        time: '1 день назад',
-        author: '',
-      },
-      id: 'e7orkus6qo',
-    },
-    {
-      title: 'Люблю вашу игру',
-      author: 'Piligrim',
-      answers: 1,
-      lastPublic: {
-        time: '2 дня назад',
-        author: 'Техническая поддержка',
-      },
-      id: '43so7tojfk',
-    },
-  ];
+  const { idTopicList } = useParams<string>();
+
+  const [topics, setTopics] = useState<ITopicTypes[]>([]);
+  const [isNewTopicOpen, setIsNewTopicOpen] = useState<boolean>(false);
 
-  const { idTopicList } = useParams();
-  const [topic] = useState<ITopicTypes[]>(testTopicList);
-  const [isNewTopicOpen, setIsNewTopicOpen] = useState(false);
+  useEffect(() => {
+    if (idTopicList != undefined) {
+      forumApi
+        .getAllTopics(idTopicList)
+        .then((topicsList) => setTopics(topicsList.data))
+        .catch((e) => console.log(e));
+    }
```

Comment:
Старайся написать код как можно плавнее (то есть уменьшать уровень вложенности).
```suggestion
    if (!idTopicList) {
       return;
    }
    
    forumApi
        .getAllTopics(idTopicList)
        .then((topicsList) => setTopics(topicsList.data))
        .catch((e) => console.log(e));
```
https://learn.javascript.ru/coding-style#urovni-vlozhennosti

Reply:
Узнал это на обучении, но пока забываю это использовать